tests/unit/test_relation_monitors.py

```diff
@@ -97,11 +97,19 @@ def test_prometheus(self):
         rel_id_prom = self.harness.add_relation("downstream-prometheus-scrape", "prom")
         self.harness.add_relation_unit(rel_id_prom, "prom/0")
 
-        # THEN alert rules are transferred to prometheus over relation data
+        # THEN alert rules are transferred to prometheus over relation data and
+        # nagios_host_context is not part of the expr.
         app_data = self.harness.get_relation_data(rel_id_prom, "cos-proxy")
-
         self.assertIn("alert_rules", app_data)  # pyright: ignore
 
+        groups = json.loads((app_data["alert_rules"]))["groups"]
+
+        for group in groups:
+            for rule in group["rules"]:
+                self.assertNotIn("my-nagios-host-context", rule["expr"])
```

Comment:
Nit: we could extract `my-nagios-host-context` to be a constant.

Reply:
Good catch!